core/interfaces.go

```diff
@@ -108,7 +108,7 @@ type CertificateAuthority interface {
 type PolicyAuthority interface {
 	WillingToIssue(domain AcmeIdentifier) error
 	WillingToIssueWildcard(domain AcmeIdentifier) error
-	ChallengesFor(domain AcmeIdentifier) (challenges []Challenge, validCombinations [][]int, err error)
+	ChallengesFor(domain AcmeIdentifier) (challenges []Challenge, err error)
```

Comment:
```suggestion
	ChallengesFor(domain AcmeIdentifier) ([]Challenge, error)
```

(Unless there's some reason that this interface needs named return parameters I'm missing)